#! /bin/sh /usr/share/dpatch/dpatch-run
## DP: Fix compilation warning (error) on SPARC, where timeval
## DP: attributes don't seem to fit "%ld".  Also format them as unsigned
## DP: numbers.

--- conntrack-tools/src/conntrack.c
+++ conntrack-tools/src/conntrack.c
@@ -597,7 +597,8 @@
 		if (!(output_mask & _O_XML)) {
 			struct timeval tv;
 			gettimeofday(&tv, NULL);
-			printf("[%-8ld.%-6ld]\t", tv.tv_sec, tv.tv_usec);
+			printf("[%-8lu.%-6lu]\t", (unsigned long)tv.tv_sec,
+                               (unsigned long)tv.tv_usec);
 		} else
 			output_flags |= NFCT_OF_TIME;
 	}	
